### PR TITLE
[T3A3][F10-A2]Shi Xiyue

### PR DESCRIPTION
**Pull request by shixiyue**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/commands/CommandResult.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 public class CommandResult {
 
     /** The feedback message to be shown to the user. Contains a description of the execution result */
-    public final String feedbackToUser;
+    private final String feedbackToUser;
```
Comment (Candiie) [inline src/seedu/addressbook/commands/CommandResult.java]: Hi Xi Yue, I do believe that this is a task in Activity T3A2 Encapsulation, do take note to only push related materials with accordance to the tasked activity in the future :) No worries for this time round!

File: src/seedu/addressbook/ui/TextUi.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public void showResultToUser(CommandResult result) {
         if(resultPersons.isPresent()) {
             showPersonListView(resultPersons.get());
         }
-        showToUser(result.feedbackToUser, DIVIDER);
+        showToUser(result.getFeedbackToUser(), DIVIDER);
```
Comment (Candiie) [inline src/seedu/addressbook/ui/TextUi.java]: Do refer to comment above :)

File: doc/UserGuide.md [hunk 3]
```diff
@@ -42,11 +42,13 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
...
 > be seen using the `viewall` command.
 > 
+> A valid address is entered in the following format a/BLOCK, STREET, UNIT, POSTAL_CODE
+>
 > Persons can have any number of tags (including 0)
 
 Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
+* `add John Doe p/98765432 e/johnd@gmail.com a/block 123, John street, #01-01, 111111`
```
Comment (Candiie) [inline doc/UserGuide.md]: Well done on updating the user guide! However, this is considered as a significant change and most importantly, you wouldn't want to commit your user guide together with your actual java implementations as it confuses the evaluator ~ do try to take note of this in the future :)

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -8,11 +8,20 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "An valid address is in the following format BLOCK, STREET, UNIT, POSTAL_CODE";
+    
+    private static final int ADDRESS_NUMBER_OF_PARTS = 4;
```
Comment (Candiie) [inline src/seedu/addressbook/data/person/Address.java]: Nice job extracting the magic numbers!

File: test/expected.txt [hunk 5]
```diff
@@ -8,7 +8,7 @@
 || Enter command: || [Command entered:  sfdfd]
 || add: Adds a person to the address book. Contact details can be marked private by prepending 'p' to the prefix.
 || Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...
-|| Example: add John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney
+|| Example: add John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25, 231534 t/friends t/owesMoney
```
Comment (Candiie) [inline test/expected.txt]: Great job on updating the test  Xi Yue  !

File: test/input.txt [hunk 6]
```diff
@@ -40,19 +40,19 @@
...
-  add Dickson Ee p/444444 pe/dickson@nus.edu.sg a/444, delta street t/friends
+  add Dickson Ee p/444444 pe/dickson@nus.edu.sg a/444, delta street, #44-44, 444444 t/friends
   list
-  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street t/tubers t/starchy
+  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street, #55-55, 555555 t/tubers t/starchy
   list
 
   # should not allow adding duplicate persons
-  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street t/tubers t/starchy
+  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street, #55-55, 555555 t/tubers t/starchy
```
Comment (Candiie) [inline test/input.txt]: Great effort !!! Well done!!!

**Comment by Candiie** [comment]
Some comments added. Please close the PR after reading comments:

:+1: Hi Xi Yue, Well done, simple and clean execution, however for this round of execution since this task is simple it is ok to do the changes and push all in 1 commit however, in future commits do take note to commit after each significant changes instead of all in one :)

File: src/seedu/addressbook/commands/CommandResult.java (hunk 1 above)
Comment (shixiyue) [inline src/seedu/addressbook/commands/CommandResult.java]: Ok thanks!

**Pull request closed without merging**